tests/extension/oneapi_non_uniform_groups/ballot_group_api.cpp

```diff
@@ -0,0 +1,175 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2024 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+namespace non_uniform_groups::tests {
+
+struct checks {
+  enum {
+    get_group_id,
+    get_local_id,
+    get_group_range,
+    get_local_range,
+    get_group_linear_id,
+    get_local_linear_id,
+    get_group_linear_range,
+    get_local_linear_range,
+    leader,
+    COUNT,
+  };
+};
+
+TEST_CASE("Test for ballot_group apis.", "[oneapi_non_uniform_groups]") {
+#ifndef SYCL_EXT_ONEAPI_NON_UNIFORM_GROUPS
+  SKIP("SYCL_EXT_ONEAPI_NON_UNIFORM_GROUPS is not defined");
+#else
+  namespace oneapi_ext = sycl::ext::oneapi::experimental;
+  using ballot_group_t = oneapi_ext::ballot_group<sycl::sub_group>;
+  using CheckResults = bool[checks::COUNT];
+
+  constexpr size_t num_items = 64;
+
+  sycl::buffer<CheckResults, 1> results_buffer{num_items};
+
+  auto q = sycl_cts::util::get_cts_object::queue();
+
+  if (!q.get_device().has(sycl::aspect::ext_oneapi_ballot_group)) {
+    SKIP("Device does not support ballot_group.");
+  }
+
+  q.submit([&](sycl::handler& cgh) {
+    sycl::accessor acc{results_buffer, cgh, sycl::write_only};
+
+    sycl::nd_range<1> nd_range{sycl::range<1>{num_items},
+                               sycl::range<1>{num_items}};
+
+    cgh.parallel_for<struct BallotGroupAPI>(nd_range, [=](sycl::nd_item<1> it) {
+      auto& results = acc[it.get_global_id()];
+
+      sycl::sub_group sg = it.get_sub_group();
+      size_t sub_group_size = sg.get_local_range().size();
+      size_t split = sub_group_size / 3;
+      bool is_left = sg.get_local_linear_id() < split;
+
+      auto ballot = oneapi_ext::get_ballot_group(sg, is_left);
```

Comment:
Should this be checked explicitly?
```
Constraints: Available only if `sycl::is_group_v<std::decay_t<Group>> &&
std::is_same_v<Group, sycl::sub_group>` is true.
```

Reply:
I don't believe the CTS normally does negative testing for availability like this. The test will fail if it isn't correctly available.